app/models/admin_post.rb

```diff
@@ -68,10 +70,16 @@ def tag_list=(list)
 
   def translated_post_must_exist
     if translated_post_id.present? && AdminPost.find_by(id: translated_post_id).nil?
-      errors.add(:translated_post_id, 'does not exist')
+      errors.add(:translated_post_id, "does not exist")
     end
   end
 
+  def translated_post_language_must_differ
+    return unless translated_post_id.present? && !AdminPost.find_by(id: translated_post_id).nil? && self.language_id == AdminPost.find_by(id: translated_post_id).language_id
```

Comment:
I think we can make this a little easier to read by breaking it up across two lines and simplifying it:
```
return unless translated_post.present?
return unless translated_post.language == language
```